applications/luci-app-filemanager/htdocs/luci-static/resources/view/system/filemanager.js

```diff
@@ -0,0 +1,1980 @@
+	windowSizes: {
+		width: 800,
+		height: 400
+	},
+	otherSettings: {} // Additional settings
+};
+
+// Function to upload a file to the server
+function uploadFile(filename, filedata, onProgress) {
+	return new Promise(function(resolve, reject) {
+		var formData = new FormData();
+		formData.append('sessionid', rpc.getSessionID()); // Add session ID
+		formData.append('filename', filename); // File name
+		formData.append('filedata', filedata); // File data
+
+		var xhr = new XMLHttpRequest();
+		xhr.open('POST', L.env.cgi_base + '/cgi-upload', true); // Configure the request
+
+		// Monitor upload progress
+		xhr.upload.onprogress = function(event) {
+			if (event.lengthComputable && onProgress) {
+				var percent = (event.loaded / event.total) * 100;
+				onProgress(percent); // Call the progress callback with percentage
+			}
+		};
+
+		// Handle request completion
+		xhr.onload = function() {
+			if (xhr.status === 200) {
+				resolve(xhr.responseText); // Upload successful
+			} else {
+				reject(new Error(xhr.statusText)); // Upload error
+			}
+		};
+
+		// Handle network errors
+		xhr.onerror = function() {
+			reject(new Error('Network error'));
+		};
+
+		xhr.send(formData); // Send the request
+	});
+}
+
+// Function to load settings from the configuration file
+function loadConfig() {
+	return fs.read(configFilePath).then(function(content) {
+		var lines = content.trim().split('\n');
+		lines.forEach(function(line) {
+			if (line.includes("option")) {
+				var splitLines = line.split("option").filter(Boolean);
+				splitLines.forEach(function(subline) {
+					subline = "option " + subline.trim();
+					var parts = subline.match(/^option\s+(\S+)\s+'([^']+)'$/);
+					if (parts && parts.length === 3) {
+						var key = parts[1];
+						var value = parts[2];
+						if (key === 'columnWidths' || key === 'columnMinWidths' || key === 'columnMaxWidths') {
+							// Parse column width settings
+							var widths = value.split(',');
+							widths.forEach(function(widthStr) {
+								var widthParts = widthStr.split(':');
+								if (widthParts.length === 2) {
+									config[key][widthParts[0]] = parseInt(widthParts[1], 10);
+								}
+							});
+						} else if (key === 'currentDirectory') {
+							config.currentDirectory = value; // Set current directory
+						} else if (key === 'windowSizes') {
+							// Parse window size settings
+							var sizes = value.split(',');
+							sizes.forEach(function(sizeStr) {
+								var sizeParts = sizeStr.split(':');
+								if (sizeParts.length === 2) {
+									var sizeKey = sizeParts[0];
+									var sizeValue = parseInt(sizeParts[1], 10);
+									if (!isNaN(sizeValue)) {
+										config.windowSizes[sizeKey] = sizeValue;
+									}
+								}
+							});
+						} else {
+							config[key] = value; // Other settings
+						}
+					}
+				});
+			}
+		});
+	}).catch(function(err) {
+		console.error("Failed to load config: " + err);
+	});
+}
```

Comment:
There may be a number of ways to improve key+value handling, but how about the following, then it's a bit less intense, using a switch case to clean up the ifs:
```js

function parseKeyValuePairs(input, delimiter, callback) {
    const pairs = input.split(',');
    pairs.forEach((pair) => {
        const [key, value] = pair.split(delimiter);
        if (key && value) callback(key.trim(), value.trim());
    });
}

async function loadConfig() {
    try {
        const content = await fs.read(configFilePath);
        const lines = content.trim().split('\n');

        lines.forEach((line) => {
            if (!line.includes('option')) return;

            const splitLines = line.split('option').filter(Boolean);

            splitLines.forEach((subline) => {
                const formattedLine = "option " + subline.trim();
                const match = formattedLine.match(/^option\s+(\S+)\s+'([^']+)'$/);

                if (!match) return;

                const [, key, value] = match;

                switch (key) {
                    case 'columnWidths':
                    case 'columnMinWidths':
                    case 'columnMaxWidths':
                        parseKeyValuePairs(value, ':', (k, v) => {
                            config[key] = config[key] || {};
                            config[key][k] = parseInt(v, 10);
                        });
                        break;

                    case 'currentDirectory':
                        config.currentDirectory = value;
                        break;

                    case 'windowSizes':
                        parseKeyValuePairs(value, ':', (k, v) => {
                            config.windowSizes = config.windowSizes || {};
                            const sizeValue = parseInt(v, 10);
                            if (!isNaN(sizeValue)) {
                                config.windowSizes[k] = sizeValue;
                            }
                        });
                        break;

                    default:
                        config[key] = value;
                }
            });
        });
    } catch (err) {
        console.error('Failed to load config: ' + err.message);
    }
}
```

Reply:
Implemented